In [ ]:
import importlib
import utils 

import pandas as pd
import numpy as np

import matplotlib.pylab as plt
from pandas.plotting import scatter_matrix

%matplotlib inline

In [ ]:
importlib.reload(utils)

_pd = utils.PrepareData()
df = _pd.df

df.head()

In [18]:
mddf = pd.DataFrame({
    'column':          [column for column in df.columns],
    'kind':            ['categorical' for column in df.columns],
    'data type':       [df[column].dtype for column in df.columns],
    'unique items':    [len(df[column].unique()) for column in df.columns],
    'nans':            [df[df[column].isnull()].shape[0] for column in df.columns],
})
print(mddf.to_markdown())

|    | column              | kind        | data type   |   unique items |   nans |
|---:|:--------------------|:------------|:------------|---------------:|-------:|
|  0 | id                  | categorical | int64       |          21436 |      0 |
|  1 | date                | categorical | object      |            372 |      0 |
|  2 | price               | categorical | float64     |           3625 |      0 |
|  3 | price_bin           | categorical | int64       |              2 |      0 |
|  4 | bedrooms            | categorical | int64       |             13 |      0 |
|  5 | bathrooms           | categorical | float64     |             30 |      0 |
|  6 | sqft_living         | categorical | int64       |           1038 |      0 |
|  7 | sqft_lot            | categorical | int64       |           9782 |      0 |
|  8 | floors              | categorical | float64     |              6 |      0 |
|  9 | waterfront          | categorical | int64       |              2 |      0 |
| 10

|    | column              | type    |   item count |   nans |\n|---:|:--------------------|:--------|-------------:|-------:|\n|  0 | Unnamed: 0          | int64   |        21613 |      0 |\n|  1 | date                | object  |          372 |      0 |\n|  2 | price               | float64 |         3625 |      0 |\n|  3 | price_bin           | int64   |            2 |      0 |\n|  4 | bedrooms            | int64   |           13 |      0 |\n|  5 | bathrooms           | float64 |           30 |      0 |\n|  6 | sqft_living         | int64   |         1038 |      0 |\n|  7 | sqft_lot            | int64   |         9782 |      0 |\n|  8 | floors              | float64 |            6 |      0 |\n|  9 | waterfront          | int64   |            2 |      0 |\n| 10 | view                | int64   |            5 |      0 |\n| 11 | condition           | int64   |            5 |      0 |\n| 12 | grade               | int64   |           12 |      0 |\n| 13 | sqft_above          | int64   |          946 |      0 |\n| 14 | sqft_basement       | int64   |          306 |      0 |\n| 15 | yr_built            | int64   |          116 |      0 |\n| 16 | yr_renovated        | int64   |           70 |      0 |\n| 17 | zipcode             | int64   |           70 |      0 |\n| 18 | lat                 | float64 |         5034 |      0 |\n| 19 | long                | float64 |          752 |      0 |\n| 20 | sqft_living15       | int64   |          777 |      0 |\n| 21 | sqft_lot15          | int64   |         8689 |      0 |\n| 22 | criminal_activities | int64   |         1625 |      0 |